lib/ansible/modules/cloud/amazon/elb_target_group.py

```diff
@@ -0,0 +1,527 @@
+    matcher:
+        description: The HTTP codes to use when checking for a successful response from a target.
+        type: dict
+        sample: {
+            "http_code": "200"
+        }
+    load_balancer_arns:
+        description: The Amazon Resource Names (ARN) of the load balancers that route traffic to this target group.
+        type: list
+        sample: []
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, camel_dict_to_snake_dict, ec2_argument_spec
+import time
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_target_group(connection, module):
+
+    try:
+        return connection.describe_target_groups(Names=[module.params.get("name")])['TargetGroups'][0]
+    except (ClientError, NoCredentialsError) as e:
+        if e.response['Error']['Code'] == 'TargetGroupNotFound':
+            return None
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+
+def diff_list(a, b):
+    """Find the entries in list a that are not in list b"""
+    b = set(b)
+    return [aa for aa in a if aa not in b]
+
+
+def wait_for_status(connection, module, target_group_arn, targets, status):
+    polling_increment_secs = 5
+    max_retries = (module.params.get('wait_timeout') / polling_increment_secs)
+    status_achieved = False
+
+    for x in range(0, max_retries):
+        try:
+            response = connection.describe_target_health(TargetGroupArn=target_group_arn, Targets=targets)
+            if response['TargetHealthDescriptions'][0]['TargetHealth']['State'] == status:
+                status_achieved = True
+                break
+            else:
+                time.sleep(polling_increment_secs)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    result = response
+    return status_achieved, result
+
+
+def create_or_update_target_group(connection, module):
+
+    changed = False
+    params = dict()
+    params['Name'] = module.params.get("name")
+    params['Protocol'] = module.params.get("protocol").upper()
+    params['Port'] = module.params.get("port")
+    params['VpcId'] = module.params.get("vpc_id")
+
+    # If health check path not None, set health check attributes
+    if module.params.get("health_check_path") is not None:
+        params['HealthCheckPath'] = module.params.get("health_check_path")
+
+        if module.params.get("health_check_protocol") is not None:
+            params['HealthCheckProtocol'] = module.params.get("health_check_protocol").upper()
+
+        if module.params.get("health_check_port") is not None:
+            params['HealthCheckPort'] = str(module.params.get("health_check_port"))
+
+        if module.params.get("health_check_interval") is not None:
```

Comment:
I've removed the defaults because they are not a good idea - if someone didn't specify the parameter but changed it via the AWS console then the module would end up changing it.